src/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -26,64 +25,66 @@ public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSett
                 {
-                    profileObject = profilesObject[profileName] as JObject;
-
-                    if (profileObject == null)
+                    if (!profilesObject.TryGetProperty(profileName, out profileObject) || profileObject.Type != JsonValueType.Object)
                     {
                         return new LaunchSettingsApplyResult(false, LocalizableStrings.LaunchProfileIsNotAJsonObject);
                     }
                 }
 
-                if (profileObject == null)
+                if (profileObject.Type == default)
                 {
-                    foreach (var prop in profilesObject.Properties())
+                    foreach (var prop in profilesObject.EnumerateObject())
                     {
-                        var profile = prop.Value as JObject;
-
-                        if (profile != null)
+                        if (prop.Value.Type == JsonValueType.Object)
                         {
-                            var cmdName = profile[CommandNameKey]?.Value<string>();
-                            if (_providers.ContainsKey(cmdName))
+                            if (prop.Value.TryGetProperty(CommandNameKey, out var commandNameElement) && commandNameElement.Type == JsonValueType.String)
                             {
-                                profileObject = profile;
-                                break;
+                                if (_providers.ContainsKey(commandNameElement.GetString()))
+                                {
+                                    profileObject = prop.Value;
+                                    break;
+                                }
                             }
                         }
                     }
                 }
 
-                var commandName = profileObject?[CommandNameKey]?.Value<string>();
+                if (profileObject.Type == default)
+                {
+                    return new LaunchSettingsApplyResult(false, LocalizableStrings.UsableLaunchProfileCannotBeLocated);
+                }
 
-                if (profileObject == null)
+                if (!profileObject.TryGetProperty(CommandNameKey, out var finalCommandNameElement)
+                    || finalCommandNameElement.Type != JsonValueType.String)
                 {
                     return new LaunchSettingsApplyResult(false, LocalizableStrings.UsableLaunchProfileCannotBeLocated);
                 }
 
+                string commandName = finalCommandNameElement.GetString();
                 if (!TryLocateHandler(commandName, out ILaunchSettingsProvider provider))
                 {
                     return new LaunchSettingsApplyResult(false, string.Format(LocalizableStrings.LaunchProfileHandlerCannotBeLocated, commandName));
                 }
 
                 return provider.TryApplySettings(model, profileObject, ref command);
             }
-            catch (JsonException ex)
+            catch (JsonReaderException ex)
```

Comment:
when type mismatch it is "InvalidOperationException"
like "42" vs 42 you may also catch that

Reply:
Btw @ahsonkhan why is that?

Reply:
That's unfortunate

Reply:
The normal transport for an integral value would seem to be JSON Number.  None of the value methods on JsonElement do type coercion, they each only work for one particular value type.  (Partly it's because as a struct cursor type it doesn't have type hierarchy to hide what things aren't available)